### PR TITLE
feat: mobile nav bugs

### LINKED ISSUE #504: Side nav needs mobile outside click to close and needs to close on navigation
Reference https://github.com/carbon-design-system/carbon-platform/pull/596 for the required changes to close this issue.

### Browser

_No response_

### Description

Comparing to https://www.carbondesignsystem.com:

Side Nav: 
- [x] We need to close the mobile navigation when you click outside of the menu (the backdrop)
- [x] We need to close the side nav on mobile when you click any of the links in the side nav
- [x] With a short viewport, we can't scroll to the bottom of the side nav

Library Side Nav: 
- [x] We need to close the mobile navigation when you click outside of the menu (the backdrop)
- [x] We need to close the side nav on mobile when you click any of the links in the side nav
- [x] With a short viewport, we can't scroll to the bottom of the side nav

### Steps to reproduce

image
image

### PR DESCRIPTION
**Pull request by andreancardona**
Closes #504 

Side Nav: 
- [x] We need to close the mobile navigation when you click outside of the menu (the backdrop)
- [x] We need to close the side nav on mobile when you click any of the links in the side nav
- [x] With a short viewport, we can't scroll to the bottom of the side nav

Library Side Nav: 
- [x] We need to close the mobile navigation when you click outside of the menu (the backdrop)
- [x] We need to close the side nav on mobile when you click any of the links in the side nav
- [x] With a short viewport, we can't scroll to the bottom of the side nav

## TESTING: 

- test side nav at `sm` & `md` breakpoints: http://localhost:3000/assets/components
- test lib side nav at `sm` & `md` breakpoints: http://localhost:3000/assets/carbon-charts

### CODE REVIEW

**Review by francinelucca — CHANGES_REQUESTED** [review]
LGTM! minor comment

**Review by alisonjoseph — APPROVED** [review]
LGTM!

File: package-lock.json [hunk 1]
```diff
@@ -41128,4 +41128,4 @@
       "integrity": "sha512-NM8/P9n3XjXhIZn1lLhkFaACTOURQXjWhV4BA/RnOv8xvgqtqpAX9IO4mRQxSx1Rlo4tqzeqb0sOlruaOy3dug=="
     }
   }
-}
+}
```
Comment (alisonjoseph) [inline package-lock.json:41131]: is this intentional?

**Review by francinelucca — APPROVED** [review]
💃

**Pull request merged**